src/xmipp/libraries/reconstruction/resolution_fso.cpp

```diff
@@ -0,0 +1,1876 @@
+					m1sizeX, m1sizeY, m1sizeZ, rot, tilt, ang_con, k);
+	    	}
+
+			MetaData mdRes;
+			fnmd = formatString("fscDirection_%i.xmd", k);
+			fnmd = fn_fscmd_folder + fnmd;
+			saveFSCToMetadata(mdRes, freq, fsc, fnmd);
+
+			anistropyParameter(fsc, directionAnisotropy, k, aniParam, thrs);
+
+			//TODO: integrate this function in FSCDIR
+			interpolationCoarse(fsc, angles,
+					freq_fourier_x, freq_fourier_y, freq_fourier_z,
+		    		threeD_FSC, counterMap,
+					freqMap, freq,
+					0.5, m1sizeX, m1sizeY, m1sizeZ,
+					rot, tilt, ang_con);
+    	}
+
+    	std::cout << "----- Directional resolution estimated -----" <<  std::endl;
+    	std::cout << "   " <<  std::endl;
+    	std::cout << "Preparing results ..." <<  std::endl;
+
+    	//ANISOTROPY CURVE
+    	aniParam /= (double) angles.mdimx;
+    	MetaData mdani;
+		saveAnisotropyToMetadata(mdani, freq, aniParam, fn_ani);
+
+		//HALF 3DFSC MAP
+		FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(threeD_FSC)
+		{
+			DIRECT_MULTIDIM_ELEM(threeD_FSC, n) /= DIRECT_MULTIDIM_ELEM(counterMap, n);
+			if (std::isnan(DIRECT_MULTIDIM_ELEM(threeD_FSC, n)) == 1)
+				DIRECT_MULTIDIM_ELEM(threeD_FSC, n) = 1.0;
+		}
+
+
+
+		//This code fix the empty line line in Fourier space
+		size_t auxVal;
+		auxVal = YSIZE(threeD_FSC)/2;
+    	long n=0;
+    	for(size_t k=0; k<ZSIZE(threeD_FSC); ++k)
+    	{
+    		for(size_t i=0; i<YSIZE(threeD_FSC); ++i)
+    		{
+    			for(size_t j=0; j<XSIZE(threeD_FSC); ++j)
+    			{
+    				if ((j == 0) && (i>auxVal))
+    					{
+    					DIRECT_A3D_ELEM(threeD_FSC,k,i,j) = DIRECT_A3D_ELEM(threeD_FSC,k,i,j+1);
+    					}
+   					++n;
+    			}
+    		}
+    	}
+
+		//DIRECTIONAL FILTERED MAP
+		MultidimArray<double> filteredMap;
+		directionalFilter(FT1, threeD_FSC, filteredMap, m1sizeX, m1sizeY, m1sizeZ);
+
+		Image<double> saveImg2;
+		saveImg2() = filteredMap;
+		saveImg2.write(fn_fscmd_folder+"filteredMap.mrc");
+
+		//FULL 3DFSC MAP
+		FileName fn;
+		createFullFourier(threeD_FSC, fn_3dfsc, m1sizeX, m1sizeY, m1sizeZ);
+
+		//SPHERE FREQUENCY REFERENCE
+		MultidimArray<double> sphere;
+		sphere.resizeNoCopy(counterMap);
+		createfrequencySphere(sphere,
+				freq_fourier_x, freq_fourier_y, freq_fourier_z);
+
+		fn = fn_fscmd_folder+"sphere.mrc";
+		//TODO: Include this in the frequency creation
+		createFullFourier(sphere, fn, m1sizeX, m1sizeY, m1sizeZ);
+
+		//DIRECTIONAL RESOLUTION DISTRIBUTION
+		fn = fn_fscmd_folder+"Resolution_Distribution.xmd";
+		resolutionDistribution(resDirFSC, fn);
+
+		std::cout << "-------------Finished-------------" << std::endl;
+    }
+
+
+//TODO: Used but to check if usefull
+
+void estimateSSNR(MultidimArray<double> &half1, MultidimArray<double> &half2,
+		int m1sizeX, int m1sizeY, int m1sizeZ)
+{
+	MultidimArray<double> noise, signal;
+	signal = half1 + half2;
+	noise = half1 - half2;
+	FourierTransformer signaltransformer(FFTW_BACKWARD), noisetransformer(FFTW_BACKWARD);
```

Comment:
multithreading

Reply:
It is just only one transform, I prefer to avoid problems, and the method is fast enough